src/halmos/__main__.py

```diff
@@ -91,13 +97,13 @@ def parse_args(args=None) -> argparse.Namespace:
     group_solver.add_argument('--solver-fresh', action='store_true', help='run an extra solver with a fresh state for unknown')
     group_solver.add_argument('--solver-subprocess', action='store_true', help='run an extra solver in subprocess for unknown')
     group_solver.add_argument('--solver-parallel', action='store_true', help='run assertion solvers in parallel')
-    group_solver.add_argument('--solver-parallel-cores', default=os.cpu_count(), type=int, help='max number of cores to use for parallel assertion solvers (default: %(default)s)')
```

Comment:
I ended up removing it for a couple reasons:

- not sure if there's a good use case for limiting the number (I certainly would like 100% usage if I can)
- the interplay between processes and threads makes this somewhat meaningless? at least much harder to guarantee 
- pools can be instantiated in a different module without requiring arg parsing

Reply:
ok, let's come back to this later if needed.